src/components/tl/mlx5/mcast/tl_mlx5_mcast_helper.h

```diff
@@ -0,0 +1,320 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5.h"
+
+ucc_status_t ucc_tl_probe_ip_over_ib(const char* ib_dev_list, struct
```

Comment:
why this function is declared here?

Reply:
this function is going to be defined in future PRs inside helper.c. It is part of out helper functions that are required for setting up the mcast group

Reply:
@samnordmann sure i will more it there

Reply:
I am not sure to understand. `MCAST_H_` is just a header guard, what does it change to declare it inside its scope?
If it is not related to mcast, do you think we should declare it in another header?

Reply:
Sure I will remove it from here and I will add it to another file. I will do it in the future PRs.